### PR TITLE
Renaming 8.21 -> 9.0

### PR DESCRIPTION
**Pull request by proux01**
Since we decided next version will be 9.0 and not 8.21

### CODE REVIEW

**Comment by proux01** [comment]
@JasonGross apparently coq-scripts will need some update (for context it has been decided that next release will be Rocq 9.0)

**Comment by JasonGross** [comment]
The error it's running into is that the auto creation of compat files is failing because it doesn't know what comes before 9.0.  We just need to make the relevant files manually and then it should keep working going forwards.

**Comment by proux01** [comment]
Great, could you please ping me when it's done.

**Comment by Zimmi48** [comment]
> auto creation of compat files

Is it related to #19370 then?

**Comment by JasonGross** [comment]
@proux01 https://github.com/mit-plv/fiat/pull/114 should take care of the issue (though I haven't tested)



> > auto creation of compat files
> 
> Is it related to #19370 then?

No, it's https://github.com/coq/coq/pull/17685#issuecomment-1604224598.  Manual intervention will still be required when Coq bumps its major version, but I think that happens infrequently enough that it should be fine.  (And the manual intervention is just "run `./bump-versions.sh` in `compatibility/` and then the equivalent of `sed 's/8_21/9_0/g; s/8\.21/9\.0/g'` and the equivalent `git mv` commands")

**Comment by proux01** [comment]
> @proux01 [mit-plv/fiat#114](https://github.com/mit-plv/fiat/pull/114) should take care of the issue (though I haven't tested)

Thanks, according to CI, this works.

FWIW there has been a discussion on whether the future versions should be Rocq 9.1, 9.2,... or 10, 11,... I don't think any decision was taken yet.

Anyway, this is ready.

**Comment by Zimmi48** [comment]
> FWIW there has been a discussion on whether the future versions should be Rocq 9.1, 9.2,... or 10, 11,... I don't think any decision was taken yet.

Actually, a decision was taken. I was the one defending to bump the major number more often, but there was a consensus against. So, while we should probably not wait for 9.20 before moving on to 10.0, this won't happen very frequently either.

File: doc/stdlib/index-list.html.template [hunk 1]
```diff
@@ -705,7 +705,7 @@ through the <tt>Require Import</tt> command.</p>
     theories/Compat/Coq818.v
     theories/Compat/Coq819.v
     theories/Compat/Coq820.v
-    theories/Compat/Coq821.v
+    theories/Compat/Coq90.v
```
Comment (Zimmi48) [inline doc/stdlib/index-list.html.template:708]: Sorry that I didn't review this before Jason adapted his tools, but this should be `Rcoq90`, not `Coq90`.

File: sysinit/coqargs.ml [hunk 2]
```diff
@@ -231,7 +231,7 @@ let set_option = let open Goptions in function
   | opt, OptionAppend v -> set_string_option_append_value_gen ~locality:OptLocal opt v
 
 let get_compat_file = function
-  | "8.21" -> "Stdlib.Compat.Coq821"
+  | "9.0" -> "Stdlib.Compat.Coq90"
```
Comment (Zimmi48) [inline sysinit/coqargs.ml:234]: For now, it is easy to direct to `Rocq90` here.
Obviously, this clashes with your work in #19370, but you are already aware of this.

**Review by Zimmi48 — COMMENTED** [review]
I'm OK merging this as is if you tell me that you intend to rename the `Coq90` file to `Rocq90` in #19370.

File: tools/configure/configure.ml [hunk 3]
```diff
@@ -22,8 +22,8 @@ open CmdArgs.Prefs
 
 let (/) = Filename.concat
 
-let coq_version = "8.21+alpha"
-let vo_magic = 82099
+let coq_version = "9.0+alpha"
+let vo_magic = 90099
```
Comment (Zimmi48) [inline tools/configure/configure.ml:26]: Sorry, it looks like I had stopped my review half way.
Normally, the use of `99` is combined with the previous version number, so for 8.21, this was `82099`. Thus, I would expect `90099` to be the magic number for 9.1, and to keep the magic number unchanged in this PR.

Comment (proux01) [reply tools/configure/configure.ml]: Indeed, good point

**Review by Zimmi48 — APPROVED** [review]
Will merge when CI has run.

**Comment by Zimmi48** [comment]
@coqbot merge now

**Pull request merged**